diff --git a/3rdparty/mgm_multi/iio/iio.c b/3rdparty/mgm_multi/iio/iio.c
index 9f28831..7475497 100644
--- a/3rdparty/mgm_multi/iio/iio.c
+++ b/3rdparty/mgm_multi/iio/iio.c
@@ -511,8 +511,10 @@ int iio_type_id(size_t sample_size, bool ieeefp_sample, bool signed_sample)
 		switch(sample_size) {
 		case sizeof(float):       return IIO_TYPE_FLOAT;
 		case sizeof(double):      return IIO_TYPE_DOUBLE;
-		case sizeof(long double): return IIO_TYPE_LONGDOUBLE;
-		case sizeof(float)/2:     return IIO_TYPE_HALF;
+		#ifdef I_CAN_HAS_LONGDOUBLE
+		    case sizeof(long double): return IIO_TYPE_LONGDOUBLE;
+		#endif//I_CAN_HAS_LONGDOUBLE
+	    case sizeof(float)/2:     return IIO_TYPE_HALF;
 		default: fail("bad float size %zu", sample_size);
 		}
 	} else {
