diff --git a/3rdparty/sift/simd/LibImages/LibImages.cpp b/3rdparty/sift/simd/LibImages/LibImages.cpp
index 6acbe4e..870d80b 100644
--- a/3rdparty/sift/simd/LibImages/LibImages.cpp
+++ b/3rdparty/sift/simd/LibImages/LibImages.cpp
@@ -10,8 +10,7 @@
 #ifdef _OPENMP
 #include <omp.h>
 #endif
-#include <xmmintrin.h>
-#include <x86intrin.h>
+#include "../../sse2neon/sse2neon.h"
 #include <cmath>
 #include <algorithm>
 #include <cstring>
diff --git a/3rdparty/sift/simd/LibImages/LibImages.h b/3rdparty/sift/simd/LibImages/LibImages.h
index 9918472..c8c5db1 100644
--- a/3rdparty/sift/simd/LibImages/LibImages.h
+++ b/3rdparty/sift/simd/LibImages/LibImages.h
@@ -5,8 +5,7 @@
 #include <stdlib.h>
 #include <string>
 #include <vector>
-#include <xmmintrin.h>
-#include <x86intrin.h>
+#include "../../../sse2neon/sse2neon.h"
 
 //! Local includes
 
